src/metrics/prometheus/cluster_backup.rs

```diff
@@ -0,0 +1,125 @@
+        format!("{}_recent_requests_successful", P_PREFIX),
+        "Recent number of requests done to external storage from backup agents"
+    )
+    .unwrap();
+    static ref P_BACKUP_STATUS_TAG: HashMap<String, IntGaugeVec> = ClusterBackupTag::register(
+        format!("{}_tag", P_PREFIX).as_str(),
+        "Backup tag information"
+    );
+}
+
+impl MetricsConvertible for ClusterBackup {
+    fn to_metrics(&self, _: &[&str]) {
+        P_BACKUP_PAUSED.set(self.paused as i64);
+
+        self.total_workers.and_set(&P_BACKUP_WORKERS_TOTAL);
+        self.instances_running.and_set(&P_BACKUP_WORKERS_RUNNING);
+
+        if let Some(io) = &self.blob_recent_io {
+            P_BACKUP_RECENT_IO_BYTES_SENT.set(io.bytes_sent);
+            P_BACKUP_RECENT_IO_BYTES_PER_SECOND.set(io.bytes_per_second);
+            P_BACKUP_RECENT_REQUESTS_FAILED.set(io.requests_failed);
+            P_BACKUP_RECENT_REQUESTS_SUCCESS.set(io.requests_successful);
+        }
+
+        for (tag, backup) in &self.tags {
+            backup.set(&P_BACKUP_STATUS_TAG, &[tag.0.as_str()])
+        }
+    }
+}
+
+impl StaticMetric<IntGaugeVec> for ClusterBackupTag {
+    fn register(prefix: &str, desc: &str) -> HashMap<String, IntGaugeVec> {
+        let stat_name = &[
+            "last_restorable_behind_seconds",
+            "last_restorable_version",
+            "running_backup",
+            "running_backup_restorable",
+            "range_bytes_written",
+            "mutation_log_written_bytes",
+        ];
+        let mut metrics = HashMap::new();
+        for name in stat_name {
+            metrics.insert(
+                name.to_string(),
+                register_int_gauge_vec!(format!("{}_{}", prefix, name), desc, &["tag"],).unwrap(),
+            );
+        }
+        metrics
+    }
+    fn set(&self, metrics: &HashMap<String, IntGaugeVec>, labels: &[&str]) {
+        let stat_name = &[
+            "last_restorable_behind_seconds",
+            "last_restorable_version",
+            "running_backup",
+            "running_backup_restorable",
+            "range_bytes_written",
+            "mutation_log_written_bytes",
+        ];
+        for name in *stat_name {
+            // Safe as we know already the stat names
+            let metric = metrics.get(name).unwrap();
+            let value: i64 = match name {
+                "last_restorable_behind_seconds" => {
+                    self.last_restorable_seconds_behind.ceil() as i64
+                }
+                "last_restorable_version" => self.last_restorable_version,
+                "running_backup" => self.running_backup as i64,
+                "running_backup_restorable" => self.running_backup_is_restorable as i64,
+                "range_bytes_written" => self.range_bytes_written,
+                "mutation_log_written_bytes" => self.mutation_log_bytes_written,
+                // Impossible case
+                &_ => -1,
```

Comment:
should we warn in the exporter and continue the loop?

Reply:
Indeed, makes sense to give logs when this can happen. I'll make the update